server/src/main/kotlin/suwayomi/tachidesk/manga/impl/Page.kt

```diff
@@ -85,9 +90,12 @@ object Page {
         val chapterDir = getChapterDir(mangaId, chapterId)
         File(chapterDir).mkdirs()
         val fileName = getPageName(index)
-
-        return getImageResponse(chapterDir, fileName, useCache) {
-            source.fetchImage(tachiyomiPage).awaitSingle()
+        return if (ChapterFile.isFile(mangaId, chapterId)) {
+            ChapterFile.fetchImage(mangaId, chapterId, index)
+        } else {
+            getImageResponse(chapterDir, fileName, useCache) {
+                source.fetchImage(tachiyomiPage).awaitSingle()
+            }
```

Comment:
We have a huge problem here, online image cache and download directory are the same place, it was kind of okay to ignore it up until now because downloads and cache were basically the same thing. But we can't ignore it now that downloads are not strictly in the same format as image cache.

Also there are two additional problems here.
1- The code is making the directories so if the chapter is downloaded as cbz, the next time an image is fetched it will make the directory.
2- If a user views a chapter online and then downloads some of them(az cbz or not) they would have unexpected and possibly incomplete(partially viewed) downloads.

As a result before merging this we need a PR that separates downloads from image cache(which would go into OS temp dir)

Reply:
Point 2 is handled somewhat `suwayomi/tachidesk/manga/impl/ChapterCbz.kt:27`.

Regarding the separation of cache and downloads is there an upcoming PR for this, or should I try to do that?

Reply:
Try it.

Reply:
Marking this as resolved since we have already merged the changes in another PR